File: src/NotesExamensSeancesBundle/Entity/Examen.php
<?php

namespace NotesExamensSeancesBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Examen
 *
 * @ORM\Table(name="examen")
 * @ORM\Entity(repositoryClass="NotesExamensSeancesBundle\Repository\ExamenRepository")
 */
class Examen
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="horaire", type="datetime")
     */
    private $horaire;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="semestre", type="string", length=255)
     */
    private $semestre;

    /**
     * @return \DateTime
     */
    public function getSemestre()
    {
        return $this->semestre;
    }

    /**
     * @param \DateTime $semestre
     */
    public function setSemestre($semestre)
    {
        $this->semestre = $semestre;
    }


    /**
     * @var int
     * @ORM\ManyToOne(targetEntity="Salle")
     * @ORM\JoinColumn(name="salle",referencedColumnName="id")
     */
    private $salle;

    /**
     * @var int
     * @ORM\ManyToOne(targetEntity="Matiere")
     * @ORM\JoinColumn(name="matiere",referencedColumnName="id")
     */
    private $matiere;

    /**
     * @return int
     */
    public function getSalle()
    {
        return $this->salle;
    }

    /**
     * @param int $salle
     */
    public function setSalle($salle)
    {
        $this->salle = $salle;
    }

    /**
     * @return int
     */
    public function getMatiere()
    {
        return $this->matiere;
    }

    /**
     * @param int $matiere
     */
    public function setMatiere($matiere)
    {
        $this->matiere = $matiere;
    }

    /**
     * @return int
     */
    public function getClasse()
    {
        return $this->classe;
    }

    /**
     * @param int $classe
     */
    public function setClasse($classe)
    {
        $this->classe = $classe;
    }


    /**
     * @var int
     * @ORM\ManyToOne(targetEntity="Classe")
     * @ORM\JoinColumn(name="classe",referencedColumnName="id")
     */
    private $classe;



    /**
     * Get id
     *
     * @return int
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set horaire
     *
     * @param \DateTime $horaire
     *
     * @return Examen
     */
    public function setHoraire($horaire)
    {
        $this->horaire = $horaire;

        return $this;
    }

    /**
     * Get horaire
     *
     * @return \DateTime
     */
    public function getHoraire()
    {
        return $this->horaire;
    }
}

File: mobile/src/entities/Examen.java


package entities;
  


public class Examen {

    private int id;
    private int salle;
    private String salleName;
    private int matiere;
    private String matiereName;
    private int classe; 
    private String classeName;
    private String horaire;
    private String semestre;

    @Override
    public String toString() {
        return "\nExamen{" + "id=" + id + ", salle  "+salleName +"=" + salle + ", matiere  "+ matiereName +"=" +matiere + ", classe "+ classeName +"=" + classe + ", horaire=" + horaire + ", semestre=" + semestre + '}';
    }

    public String getSalleName() {
        return salleName;
    }

    public void setSalleName(String SalleName) {
        this.salleName = SalleName;
    }

    public String getMatiereName() {
        return matiereName;
    }

    public void setMatiereName(String matiereName) {
        this.matiereName = matiereName;
    }

    public String getClasseName() {
        return classeName;
    }

    public void setClasseName(String classeName) {
        this.classeName = classeName;
    }

    public Examen(int id, int salle, int matiere, int classe, String horaire, String semestre) {
        this.id = id;
        this.salle = salle;
        this.matiere = matiere;
        this.classe = classe;
        this.horaire = horaire;
        this.semestre = semestre;
        /*
        ExamenServices eS =  new ExamenServices();
        this.salleName = eS.getSalleName(this.salle);
        this.matiereName = eS.getMatiereName(this.matiere);
        this.classeName = eS.getClasseName(this.classe);
        */
     }

    public Examen(int salle, int matiere, int classe, String horaire, String semestre) {
        this.salle = salle;
        this.matiere = matiere;
        this.classe = classe;
        this.horaire = horaire;
        this.semestre = semestre;
    }

    public int getSalle() {
        return salle;
    }

    public void setSalle(int salle) {
        this.salle = salle;
    }

    public int getMatiere() {
        return matiere;
    }

    public void setMatiere(int matiere) {
        this.matiere = matiere;
    }

    public int getClasse() {
        return classe;
    }

    public void setClasse(int classe) {
        this.classe = classe;
    }

     
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getHoraire() {
        return horaire;
    }

    public void setHoraire(String horaire) {
        this.horaire = horaire;
    }

    public String getSemestre() {
        return semestre;
    }

    public void setSemestre(String semestre) {
        this.semestre = semestre;
    }
    
   
}